src/when-contexts.ts

```diff
@@ -0,0 +1,61 @@
+import * as vscode from 'vscode';
+import { TokenCursor } from './cursor-doc/token-cursor';
+import * as docMirror from './doc-mirror'
+
+type CursorContext = 'calva-standard' | AltCursorContext;
+type AltCursorContext = 'string' | 'comment';
+const stringContext = 'calva:cursorInString';
+const commentContext = 'calva:cursorInComment';
```

Comment:
It is a bit messy in the code base regarding this. But I mostly try to indicate this type of constant by making the variable name all uppercase.